### PR TITLE
Task 1: Create a CLI application to store results in a file and show the results

### PR DESCRIPTION
**Pull request by sandhyashahi**
# Assignment Details

https://github.com/leapfrogtechnology/lf-training/blob/master/Python/tasks/task1.md

# Completed

- [x] Store arguments in a file
- [x]  Show result in specified format

# Result screenshot
image

# Remaining

- [ ] Unit test

### CODE REVIEW

**Comment by sanjaymhj** [comment]
- Instead of keeping code in global level, consider controlling code flow from https://docs.python.org/3/library/__main__.html
- For naming conventions follow https://www.coding-guidelines.lftechnology.com/docs/python/files
- Also, Please add test cases

File: Python/Sandhya/task1/show_result.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import json
+import argparse
+from pathlib import Path
+
+#load json contents into an object
+def get_result(filename):
+    if Path.is_file(Path(Path.cwd(),filename)):
```
Comment (sagarlf) [inline Python/Sandhya/task1/show_result.py:7]: `Path` can be used as single object extension.

- use `Path(__file__).parent` for directory of file path. `Path.cwd()` would be calling path when program is called from another location.
- Use `joinpath`

File: Python/Sandhya/task1/store_result.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
+import argparse
+from datetime import datetime
+import json
+from pathlib import Path
+
+#Validate if entered date is valid or not
+#Then return date in string format
+def valid_date(date_input):
+    return datetime.strptime(date_input, "%Y/%m/%d").strftime("%Y/%m/%d")
```
Comment (sagarlf) [inline Python/Sandhya/task1/store_result.py:9]: What if I sent any garbled data as date. Error should be specified to user.
Also there is no meaning if you just do strptime and strftime with same format as human readable would be something like "2020/08/06" -> `6 Aug, 2020`

**Review by sagarlf — COMMENTED** [review]
@sandhyashahi looks good but you should see review comments and comments that @sanjaymhj made.
If you would like to change as per review will wait for a day.

File: Python/Sandhya/task1/store_result.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+#Calculate percentage
+def perc(score, total):
+    return (score / total) * 100
+
+#Creates and overwrites if exists
+def writeFile(filename, content):
```
Comment (Basanta78) [inline Python/Sandhya/task1/store_result.py:16]:
```suggestion
def write_file(filename, content):
```

File: Python/Sandhya/task1/store_result.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+args = parser.parse_args()
+
+filename = args.store
+
+values = args.__dict__
+
+
+#Delete filename information from values
+del values['store']
```
Comment (Basanta78) [inline Python/Sandhya/task1/store_result.py:48]: Deleting "store" might not be necessary here. Just an extra operation here.

File: Python/Sandhya/task1/store_result.py [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+if Path.is_file(Path(Path.cwd(),filename)):
+    with open(filename, 'r') as file1:
+        file_content = json.load(file1)
+
+        #Get index from the list, if the student info already exist
+        def get_index(file_content):
+            for num, d in enumerate(file_content):
+                if d['name'] == values['name']:
+                    index_of = num
+                    print(index_of)
```
Comment (Basanta78) [inline Python/Sandhya/task1/store_result.py:77]: Try to remove unnecessary print statements. Clean unnecessary print statements before pushing the code.

**Pull request merged**